app/workers/kms_encryption_verification_job.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class KmsEncryptionVerificationJob
+  include Sidekiq::Worker
+  include SentryLogging
+
+  def perform(models = ApplicationRecord.descendants_using_encryption.map(&:name))
+    corrupted_records = models.map(&:constantize).flat_map do |model|
+      attributes = model.lockbox_attributes.keys
+      model.where('encrypted_kms_key IS NOT NULL').flat_map do |record|
+        attributes.flat_map do |attribute|
+          record unless decrypted?(record, attribute)
+        end
+      end
+    end.compact
+    log_corrupted_records(corrupted_records) if corrupted_records.any?
+  end
+
+  private
+
+  def decrypted?(record, attribute)
+    box = KmsEncrypted::Box.new
+    decrypted_data_key = box.decrypt(record.encrypted_kms_key,
+                                     context: { model_name: record.class.name, model_id: record.id })
+    lockbox = Lockbox.new(key: decrypted_data_key, encode: true)
+    lockbox.decrypt record.public_send("#{attribute}_ciphertext")
+  rescue Lockbox::DecryptionError
+    Rails.logger.error("Record re-encryption unsuccessful model:
+                        #{record.class.name} - id: #{record.id} - attribute: #{attribute}")
+    false
+  else
+    true
+  end
+
+  def log_corrupted_records(corrupted_records)
+    log_message_to_sentry('Lockbox Decryption Error: Lockbox::DecryptionError', :error,
```

Comment:
I think just raising is enough to send it to sentry, right?

Reply:
Ah yes, I think you're right. The exception will bubble up to the application_controller (I think) and be logged to Sentry via [this concern ](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/concerns/exception_handling.rb)